gateway/src/error.rs

```diff
@@ -542,35 +400,12 @@ impl std::fmt::Display for ErrorDetails {
                     message
                 )
             }
-            ErrorDetails::FireworksClient { message, .. } => {
-                write!(f, "Error from Fireworks client: {}", message)
-            }
-            ErrorDetails::FireworksServer { message } => {
-                write!(f, "Error from Fireworks server: {}", message)
-            }
             ErrorDetails::GCPCredentials { message } => {
                 write!(f, "Error in acquiring GCP credentials: {}", message)
             }
-            ErrorDetails::GCPVertexClient { message, .. } => {
-                write!(f, "Error from GCP Vertex client: {}", message)
-            }
-            ErrorDetails::GCPVertexServer { message } => {
-                write!(f, "Error from GCP Vertex server: {}", message)
-            }
-            ErrorDetails::GoogleAIStudioClient { message, .. } => {
-                write!(f, "Error from Google AI Studio client: {}", message)
-            }
-            ErrorDetails::GoogleAIStudioServer { message } => {
-                write!(f, "Error from Google AI Studio server: {}", message)
-            }
-            ErrorDetails::HyperbolicClient { message, .. } => {
-                write!(f, "Error from Hyperbolic client: {}", message)
-            }
-            ErrorDetails::HyperbolicServer { message } => {
-                write!(f, "Error from Hyperbolic server: {}", message)
-            }
             ErrorDetails::Inference { message } => write!(f, "{}", message),
-            ErrorDetails::InferenceClient { message } => write!(f, "{}", message),
+            ErrorDetails::InferenceClient { message, .. } => write!(f, "{}", message),
+            ErrorDetails::InferenceServer { message, .. } => write!(f, "{}", message),
```

Comment:
[for discussion - @Kaboomtastic no need to update yet, cc @virajmehta]

Maybe we should add a prefix to this error message, like the old `Error from XYZ client/server"?

Potentially add a field to the error struct that contains the provider, so we can include that in the error message?

```
Error from inference [client|server] (model.model_name.provider.provider_name): error message (status code: xxx)
```

The span has quite a bit of info but the error message returned to the client doesn't.

Reply:
I think this is reasonable but shouldn't block merging. I will open another issue as I review.

Reply:
IMO we should address it because otherwise the error message will be very mysterious on the client side. If I understand this correctly, the client will only receive messages like `{\"error\":\"unauthorized\"}` - unclear if this is a gateway issue or downstream issue.

Reply:
@Kaboomtastic I actually agree with @GabrielBianconi here on second thought -- we should add a field to the InferenceClientError and InferenceServerError that contains the provider type "openai"|"anthropic"|... so that we can use this in the error message, i.e. 

```rust
 ErrorDetails::InferenceClient { message, provider_type, .. } => write!(f, "Error from {} client: {}", provider_type, message),
 ErrorDetails::InferenceServer { message, provider_type, .. } => write!(f, "Error from {} server: {}", provider_type, message),
```

this would help people understand what isn't working. Do you mind adding this?

Reply:
You don't have that information at the time the error is created. We would have to build the string at runtime and change the `provider_trait` interface to do so OR change the `ProviderConfig` struct so that it contains the string and we can get it with `self.config_path`.  This would require us to build this string at initialization time for each model provider. I think the latter is better and easier to test.  I think this is gonna be too involved for now but worth doing soon. Maybe we do the simple thing now as I suggested and then add this as an issue?

Reply:
Sounds good

Reply:
I can add the provider and model name to the error message. I don't know if it's in scope of this issue but I think there should be either some function or constant that is associated to each implementor of Provider that is used for provider names universally.

Reply:
you mean for them to render nicely? agree that that would be better, but for now let's hardcode it since we want to do paths eventually. thank you!